cmd/virt-launcher/virt-launcher.go

```diff
@@ -468,8 +468,11 @@ func main() {
 	finalShutdownCallback := func(pid int) {
 		err := domainManager.KillVMI(vmi)
 		if err != nil {
-			log.Log.Reason(err).Errorf("Unable to stop qemu with libvirt, falling back to SIGTERM")
-			syscall.Kill(pid, syscall.SIGTERM)
+			log.Log.Reason(err).Errorf("Unable to stop qemu with libvirt")
+			if pid != 0 {
+				log.Log.Warning("Falling back to SIGTERM")
+				syscall.Kill(pid, syscall.SIGTERM)
```

Comment:
I am not 100% sure about the initial intention here. But I would assume it was meant to send SIGTERM only to the Qemu process, not to virt-launcher and all its chilfren.